### PR TITLE
Migrate to new repo

### PR DESCRIPTION
**Pull request by reginleif888**

### CODE REVIEW

File: src/AccountManager.mo
Comment (timohanke) [inline src/AccountManager.mo]: The function name is close to the `process_deposit` above. 

Maybe `process_allowance` is better to distinguish it?

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: We should export these type from TokenHandler so that the example code does not have to define them.

For consistent naming better to call these NotifyRespose instead of NotifyResult.

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: It looks like we need some convenience functions to reduce this boiler plate code for the developer.

For example, we could provide a convenience function in ICRC1.mo which takes either a principal or the actor and produces the type LedgerAPI which can then be passed as the first argument to the TokenHandler constructor.

Or we need a second constructor in lib.mo which wraps around TokenHandler and which takes a principal instead of type LedgerAPI as first argument and does the rest internally.

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: If the null case leads to a `return` then a `let-else` statement is often nicer. Can't use the pipe then but still. 

The icrcX spec says to throw an error (i.e. `throw Error.reject("..")`) in case the token isn't known.

For example
```
let ?assetInfo = getAssetInfo(token) else throw Error.reject("Unknown token");
```

File: example/main.mo
Comment (timohanke) [inline example/main.mo:137]: The icrcX spec says to throw an error (i.e. `throw Error.reject("..")`) in case the token isn't known. It does not return a Result type, i.e. no #Ok/#Err variant.

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: Can use `let-else` instead.

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: Can use let-else instead

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: This line can be deleted. `handler.notify()` handles fee internally.

File: example/main.mo
Comment (timohanke) [inline example/main.mo:175]: icrcX spec says to return `#Err(#NotAvailable)` in this case.

Comment (timohanke) [inline example/main.mo]: icrcX spec says to return #Err(#NotAvailable) in this case.

File: example/main.mo
Comment (timohanke) [inline example/main.mo]: We can remove this line. TokenHandler was configured with auto-trigger = true.

File: example/main.mo
Comment (reginleif888) [reply example/main.mo]: Naming is taken from the standard: https://github.com/research-ag/icrcX/blob/main/icrcX.md#notification
Regarding types reuse, we should define them locally, because these types are not related to the token handler.

Comment (timohanke) [reply example/main.mo]: You are right. The types are related to icrcX, not to TokenHandler. So we could make a file icrcX.mo in the future to import these files.

**Review by timohanke — DISMISSED** [review]

**Review by timohanke — COMMENTED** [review]
Looks good. Just waiting for additional tasks to be merged in here before we merge this into main.

**Review by timohanke — COMMENTED** [review]
For the last commit we could use a test of`handler.depositFromAllowance(p, account, amount)` where the caller principal `p` is different from the principal in `account`. Just to test that internally the right principal gets credited.

File: .github/workflows/notify_on_tag_creation.yml
Comment (timohanke) [inline .github/workflows/notify_on_tag_creation.yml]: Can we remove these lines now because we have this at the top?
```
env:
  NOTION_DATABASE_ID: e3aca840563b4cc8b273e65da748e1dd
  WEBHOOK_URL: https://hook.eu2.make.com/sg1xhhlvnp4314llgde92csqgjrpm42e
```

File: .github/workflows/notify_on_tag_creation.yml
Comment (timohanke) [inline .github/workflows/notify_on_tag_creation.yml]: Have you tested that this tag pattern works as expected?

**Review by timohanke — COMMENTED** [review]
Left one request about notify hook

**Comment by timohanke** [comment]
Abandoning this PR in favor of #20 . Work will continue on #20

**Pull request closed without merging**
